### DISCUSSION TITLE
CommercialUnsupportedException error can be misleading

### DISCUSSION THREAD

**Discussion by bwmodular** [DynamicsValue/fake-xrm-easy#89] [General]
I was caught out by something earlier today and thought it might be useful to discuss ways to help other users avoid the same trap.

I added a test to an existing project which so far had only used the Core FakeXrmEasy packages.
My new tests were testing some code which shares a record with a user.

The test threw this exception:
_The organization request type 'Microsoft.Crm.Sdk.Messages.GrantAccessRequest' is not yet supported...etc._ 
but I _knew_ that this was already supported in Version1, so I was confused.

I worked out quite quickly that this was because I hadn't included FakeXrmEasy.Messages (and pointed to this with the AddFakeMessageExecutors directive in my FakeXrmEasyTestsBase) and was able to resolve it and get the tests running.

I know that the CommercialUnsupportedException exception message is already quite long, but it might be worth adding some additional text to say 'Have you included FakeXrmEasy.Messages' in your project. Otherwise users who are new to FXE might genuinely think that this is not implemented in v2/3 and think that the framework is far less capable than it is....

**Comment by jordimontana82** [comment]
Agree  @bwmodular . Got many people asking the same question....

**Reply by bwmodular** [reply]
I've been thinking about more helpful wording for this exception since I first started this discussion. It's tricky isn't it as sometimes the current wording is correct, and sometimes it won't be correct and you don't want to write a big long essay as part of the exception to try and explain this to people. Why don't I turn this into an Issue anyway and we can discuss further....
